pkg/artifactory/resource_artifactory_backup.go

```diff
@@ -0,0 +1,251 @@
+	SendMailOnError        bool          `xml:"sendMailOnError" yaml:"sendMailOnError"`
+}
+
+type RepositoryRef struct {
+	RepositoryRef []string `xml:"repositoryRef"`
+}
+
+type Backups struct {
+	BackupArr []BackupGet `xml:"backups>backup" yaml:"backup"`
+}
+
+func resourceArtifactoryBackup() *schema.Resource {
+	var backupSchema = map[string]*schema.Schema{
+		"key": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+			Description:      `(Required) Backup config name.`,
+		},
+		"enabled": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: `(Optional) Flag to enable or disable the backup config. Default value is "true".`,
+		},
+		"cron_exp": {
+			Type:             schema.TypeString,
+			Required:         true,
+			ValidateDiagFunc: validation.ToDiagFunc(validateCron),
+			Description:      `(Required) Cron expression to control the backup frequency.`,
+		},
+		"retention_period_hours": {
+			Type:             schema.TypeInt,
+			Optional:         true,
+			Default:          168,
+			ValidateDiagFunc: validation.ToDiagFunc(validation.IntAtLeast(0)),
+			Description:      `(Optional) The number of hours to keep a backup before Artifactory will clean it up to free up disk space. Applicable only to non-incremental backups. Default value is 168 hours ie: 7 days.`,
+		},
+		"excluded_repositories": {
+			Type:        schema.TypeList,
+			Optional:    true,
+			Elem:        &schema.Schema{Type: schema.TypeString},
+			Description: `(Optional) list of excluded repositories from the backup. Default is empty list.`,
+		},
+		"create_archive": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: `(Optional) If set to true, backups will be created within a Zip archive (Slow and CPU intensive). Default value is 'false'`,
+		},
+		"exclude_new_repositories": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: `(Optional) When set to true, new repositories will not be automatically added to the backup. Default value is 'false'.`,
+		},
+		"send_mail_on_error": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     true,
+			Description: `(Optional) If set to true, all Artifactory administrators will be notified by email if any problem is encountered during backup. Default value is 'true'.`,
+		},
+	}
+	var resourceBackupRead = func(_ context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+		backups := &Backups{}
+		backup := unpackBackup(d)
+
+		_, err := m.(*resty.Client).R().SetResult(&backups).Get("artifactory/api/system/configuration")
+		if err != nil {
+			return diag.Errorf("failed to retrieve data from API: /artifactory/api/system/configuration during Read")
+		}
+
+		matchedBackup := Backup{}
+		for _, iterBackup := range backups.BackupArr {
```

Comment:
Please make this loop take on separation of concerns. In this loop you are doing 2 things:

1. Looping to find
2. Setting values

Use the loop to just find your backup. Then the matched backup code can be flatted and removed from the for loop.

Finding something and acting upon it are entirely different:

See also
`filter`
`map`